### PR TITLE
Add options to make the planets polarized

### PR DESCRIPTION
**Pull request by keskitalo**
This PR extends the planetary emission models to include linear polarization.  The user sets the polarization fraction and angle that get applied to all planetary observations and all planets. The reference frame is set by the workflow and is currently set to be the Celestial system.

### CODE REVIEW

**Review by mhasself — CHANGES_REQUESTED** [review]
Thank you!  This seems to be working, other than my note about using 2*angle.

File: sotodlib/toast/ops/sim_sso.py [hunk 1]
```diff
@@ -412,6 +457,35 @@ def _observe_sso(
...
+            weights_U = weights[good, ind].copy()
+        else:
+            weights_U = 0
+
+        pfrac = self.polarization_fraction
+        angle = self.polarization_angle.to_value(u.radian)
+
+        sig *= weights_I + pfrac * (
+            np.cos(angle) * weights_Q + np.sin(angle) * weights_U
+        )
```
[the remark this answers is not in the document]
Comment (keskitalo) [reply sotodlib/toast/ops/sim_sso.py:485]: I am not sure I agree. The polarization angle here is used to generate an image of the polarized source. If the user sets `polarization_angle=0 deg` they expect to find `U=0` and if they set `polarization_angle=45 deg` they expect to find `Q = 0`. The polarization _weights_ do include `2 X angle`.

[the remark this answers is not in the document]
Comment (mhasself) [reply sotodlib/toast/ops/sim_sso.py:485]: It sounds like you are saying that cos(45 deg) = 0... am I completely misreading this equation?

For what it's worth, I simmed a source at angle=30 ... but measured it in the output map to have angle=15.

[the remark this answers is not in the document]
Comment (keskitalo) [reply sotodlib/toast/ops/sim_sso.py]: You are correct! Thanks for noticing. I fixed it here and also in `sim_source.py` where the incorrect model was copied from.

**Review by mhasself — APPROVED** [review]

**Pull request merged**